lib/requestIterator.js

```diff
@@ -106,9 +106,9 @@ RequestIterator.prototype.rebuildRequest = function () {
   return data
 }
 
-RequestIterator.prototype.recordBody = function (request, status, body) {
+RequestIterator.prototype.recordBody = function (request, status, body, headers) {
   if (request && typeof request.onResponse === 'function') {
-    request.onResponse(status, body, this.context)
+    request.onResponse(status, body, this.context, headers)
```

Comment:
Can you move the array-to-object logic within this if? We can do it only if `onResponse` is set.

Reply:
Sounds good! it could spare us an unnecessary conversation.